marshaling.go

```diff
@@ -137,6 +137,17 @@ func (ws *Worksheet) StructScan(dest interface{}) error {
 		return fmt.Errorf("dest must be a *struct")
 	}
 
+	// dests stores refs to any worksheets that we have already scanned
+	// for reuse (and cycle termination)
+	dests := map[string]interface{}{
+		ws.Id(): dest,
+	}
+
+	return structScan(dests, ws)
+}
+
+func structScan(dests map[string]interface{}, ws *Worksheet) error {
```

Comment:
since we don't recurse into this func, do we need to define it separately, vs simply having it in `StructScan`?